content/blog/posts/visual-studio-2022-update.md

```diff
@@ -0,0 +1,277 @@
+
+
+
+## Useful build tips
+
+### Building against the VS2022 v142 toolset using VS2022
+
+One lesser known trick to build against multiple Microsoft build tools toolsets(v143, v142, etc...) without the need to have multiple versions of Visual Studio installed involves specifying the toolset option for Visual Studio on the command line when configuring with CMake.  
+
+First the user should make sure they have the latest `MSVC v142 - VS 2019 C++ x64/x86 build tools` component installed via the Visual Studio Installer.  
+{{< image-width src="/images/blog/vs2022/vs2022-installer-v142-toolset.png" width="100%" alt="Visual Studio Installer Individual Components" >}}
+
+Once it is verified that latest `MSVC v142` toolset is installed, the user can configure the toolset using either the CMake GUI or the CMake CLI via the `-T<toolset>` option which is documented on the CMake website as part of the [CMAKE_GENERATOR_TOOLSET](https://cmake.org/cmake/help/v3.23/variable/CMAKE_GENERATOR_TOOLSET.html#visual-studio-toolset-selection) variable documentation.  
+To use the the latest v142 toolset which is 14.29 at this time, the following value for the `-T` option should be supplied:
+```
+-Tv142
+```
+
+
+#### Configuring the CMake Gui to use the v142 toolset using VS 2022
+
+Using the CMake GUI to build against the VS2019 v142 toolset is as simple as specifying v142 to the toolset parameter.  
+{{< image-width src="/images/blog/vs2022/cmake-gui-14-29-toolset.png" width="100%" alt="CMake GUI Visual Studio Toolset" >}}
+
+
+#### Configure the CMake CLI to use the v142 toolset using VS2022
+
+Building using the v142 toolset using the CLI only involves adding the `-Tv142` option.  
+This uses the 19.29 build tools.  
+```powershell
+PS C:\code\o3de> cmake -B build/windows_vs142_cli -S . -Tv142
+-- Building for: Visual Studio 17 2022
+-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19044.
+-- The C compiler identification is MSVC 19.29.30146.0
+-- The CXX compiler identification is MSVC 19.29.30146.0
+```
+
+This is juxaposed against not specifying the `-T` option at all.  
+When not specified it uses the latest v143 toolset, which at the time is 19.33.  
+```powershell
+PS C:\code\o3de> cmake -B build/windows_vs143_cli -S .
+-- Building for: Visual Studio 17 2022
+-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19044.
+-- The C compiler identification is MSVC 19.33.31630.0
+-- The CXX compiler identification is MSVC 19.33.31630.0
+```
+
+### Using O3DE CMake Presets (CMake 3.23+)
+O3DE provides a [CMakePresets.json](https://github.com/o3de/o3de/blob/main/CMakePresets.json) in the repo root, which leverages the [CMakePresets](https://cmake.org/cmake/help/v3.23/manual/cmake-presets.7.html) feature which is way to shared configuration, build and test settings with other users.  
```

Comment:
update: `feature which is way to shared configuration` -> `feature which is a way to share configuration`